internal/api/router.go

```diff
@@ -0,0 +1,80 @@
+package api
+
+import (
+	"context"
+	"github.com/Azzonya/go-shortener/internal/logger"
+	"github.com/Azzonya/go-shortener/internal/middleware"
+	shortener_service "github.com/Azzonya/go-shortener/internal/shortener"
+	"github.com/gin-gonic/gin"
+	"go.uber.org/zap"
+	"log"
+	"net/http"
+)
+
+type Rest struct {
+	server    *http.Server
+	shortener *shortener_service.Shortener
+
+	ErrorChan chan error
+}
+
+func New(shortener *shortener_service.Shortener) *Rest {
+	return &Rest{
+		shortener: shortener,
+
+		ErrorChan: make(chan error, 1),
+	}
+}
+
+func (o *Rest) Start(lAddr string) {
+	logger.Log.Info("Running server", zap.String("address", lAddr))
+
+	gin.SetMode(gin.ReleaseMode)
+
+	r := gin.Default()
+
+	r.Use(
+		middleware.RequestLogger(logger.Log),
+		middleware.CompressRequest(),
+		middleware.DecompressRequest(),
+		gin.Recovery())
+
+	o.SetRouters(r)
+
+	o.server = &http.Server{
+		Addr:    lAddr,
+		Handler: r,
+	}
+
+	go func() {
+		defer func() {
+			if rec := recover(); rec != nil {
+				log.Printf("Recovered from panic: %v", rec)
```

Comment:
```suggestion
				log.Error("Recovered from panic: %v", rec)
```
но вообще сомневаюсь что тут нужно панику ловить - это не уровень наших ручек - там ловится паники отдельно мидлварой - значит здесь скорее при старте приложения возникла ошибка. Панику поймали и вышли из горутины, не передавая панику выше. Приложение не завершится но и сервер работать не будет